factor/totp/classes/factor.php

```diff
@@ -184,19 +184,29 @@ public function login_form_definition($mform) {
     public function login_form_validation($data) {
         $factors = $this->get_active_user_factors();
         $result = array('verificationcode' => get_string('error:wrongverification', 'factor_totp'));
+        $window = get_config('factor_totp', 'window');
```

Comment:
I've just realized if you set the window to say 4, then we need to test back 4 windows in time when checking the code reuse in the other pr

Reply:
I updated it to use the window in code as well

Reply:
Im not convinced that is exactly the correct behaviour. If checking the previous timestamp, with the window, all codes in the window are treated as already used. If using a window of one, we risk missing the actual code that was used.